cli/main.rs

```diff
@@ -389,7 +390,15 @@ fn resolve_flags_and_init(
     // Using same default as VSCode:
     // https://github.com/microsoft/vscode/blob/48d4ba271686e8072fc6674137415bc80d936bc7/extensions/typescript-language-features/src/configuration/configuration.ts#L213-L214
     DenoSubcommand::Lsp => vec!["--max-old-space-size=3072".to_string()],
-    _ => vec![],
+    _ => {
+      if *DENO_FUTURE {
+        // deno_ast removes TypeScript `assert` keywords, so this flag only affects JavaScript
+        // TODO(petamoriken): Need to check TypeScript `assert` keywords in deno_ast
```

Comment:
Do you think we should address this one before landing the PR?

Reply:
Yes, I think so. It would be better to work on a separate PR for TypeScript, since it is likely that the error will be displayed on the deno_ast side (https://github.com/swc-project/swc/issues/8893#issuecomment-2080369150), rather than using this flag.

Reply:
Let me talk with @dsherret about this first 👍